### PR TITLE
[DependencyInjection] remove deprecated autowiring_types feature

### PR DESCRIPTION
**Pull request by hhamon**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | ~
| License       | MIT
| Doc PR        | ~

### CODE REVIEW

**Comment by hhamon** [comment]
Forward compatibility needed in FrameworkBundle in version 3.4.

**Comment by nicolas-grekas** [comment]
missing CHANGELOG updates in components (please check any your other PRs also if any)

**Comment by nicolas-grekas** [comment]
rebase needed

**Comment by hhamon** [comment]
@nicolas-grekas rebased! Still need to update the `CHANGELOG` file.

File: src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php [hunk 1]
```diff
@@ -370,12 +370,6 @@ private function populateAvailableType($id, Definition $definition)
             return;
         }
 
-        foreach ($definition->getAutowiringTypes(false) as $type) {
-            $this->definedTypes[$type] = true;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: this property is not used anymore, it should be removed

File: src/Symfony/Component/DependencyInjection/Definition.php [hunk 2]
```diff
@@ -792,72 +792,4 @@ public function setAutowired($autowired)
...
-     *
-     * @deprecated since version 3.3, to be removed in 4.0.
-     */
-    public function getAutowiringTypes(/*$triggerDeprecation = true*/)
-    {
-        if (1 > func_num_args() || func_get_arg(0)) {
-            @trigger_error('Autowiring-types are deprecated since Symfony 3.3 and will be removed in 4.0. Use aliases instead.', E_USER_DEPRECATED);
-        }
-
-        return array_keys($this->autowiringTypes);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Definition.php]: the property should be removed from the class definition

**Comment by nicolas-grekas** [comment]
`grep autowiring.*type src/ -ri` raises a few more occurrences

File: src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php (hunk 1 above)
Comment (hhamon) [inline src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: Are you sure? It seems the property remains used several times in this class.

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: it's read, but never written to, so yet, I'm sure :)

**Comment by hhamon** [comment]
I think it's good to go now.

The remaining files below only deal with the `autowire` option but not `autowiring_types`.

```
src//Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php
src//Symfony/Bundle/FrameworkBundle/Tests/Functional/AutowiringTypesTest.php
src//Symfony/Bundle/FrameworkBundle/Tests/Functional/Bundle/TestBundle/AutowiringTypes/AutowiredServices.php
src//Symfony/Bundle/FrameworkBundle/Tests/Functional/app/AutowiringTypes/config.yml
src//Symfony/Bundle/SecurityBundle/Tests/Functional/AutowiringTypesTest.php
src//Symfony/Bundle/SecurityBundle/Tests/Functional/app/AutowiringTypes/config.yml
```

Maybe they should be renamed at some point.

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @hhamon.

**Pull request merged**